File: Introduction_to_MPI/Exercises/fortran/ex2-pingpong/pingpong.f90
program pingpong

  use mpi
  implicit none

  integer :: ifail
  integer :: rank, size


  call mpi_init(ifail)

  call mpi_comm_rank(MPI_COMM_WORLD, rank, ifail)
  call mpi_comm_size(MPI_COMM_WORLD, size, ifail)

  if (size /= 2) then
     write (unit = *, fmt = *) "Must use exactly two MPI tasks"
     call mpi_abort(MPI_COMM_WORLD, -1, ifail)
  end if


  ! ...


  call mpi_finalize(ifail)

end program pingpong

File: Introduction_to_MPI/Exercises/c/ex2-pingpong/pingpong.c
#include <mpi.h>
#include <stdio.h>

/* Pingpong program */

int main (int argc, char* argv[]){

  int rank, size;

  /* ... */
	
  MPI_Init(&argc,&argv);

  MPI_Comm_rank(MPI_COMM_WORLD,&rank);
  MPI_Comm_size(MPI_COMM_WORLD,&size);

  if (size != 2){
  	printf("Must use exactly two MPI tasks\n");
        MPI_Abort(MPI_COMM_WORLD, 1);
  }

  /* ... */
  /* ... */

  MPI_Finalize();
  
  return 0;

} 
